src/rgw/rgw_arrow.cc

```diff
@@ -0,0 +1,97 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+
+#include <iostream>
+#include <mutex>
+#include <map>
+
+#include "arrow/type.h"
+#include "arrow/flight/server.h"
+
+#include "common/dout.h"
+
+#include "rgw_arrow.h"
+
+
+#define dout_subsys ceph_subsys_rgw
+
+#define INFO(cct) ldout(cct, 20) << __PRETTY_FUNCTION__ << ": "
+#define STATUS(cct) ldout(cct, 10) << __PRETTY_FUNCTION__ << ": "
+#define WARN(cct) ldout(cct, 0) << "WARNING: " << __PRETTY_FUNCTION__ << ": "
+#define ERROR(cct) ldout(cct, 0) << "ERROR: " << __PRETTY_FUNCTION__ << ": "
+
+// assume cct is defined
+#define INFOC INFO(cct)
+#define STATUSC STATUS(cct)
+#define WARNC WARN(cct)
+#define ERRORC ERROR(cct)
+
+namespace arw = arrow;
+namespace flt = arrow::flight;
+
+
+class RGWArrowServer;
+
+
+// Globals
+
+static std::mutex server_mtx; // manages setting of server variable
+static RGWArrowServer* server = nullptr;
+
+
+class RGWArrowServer : public flt::FlightServerBase {
+
+  // types / aliases
+
+  using Data1 = std::vector<std::shared_ptr<arw::RecordBatch>>;
+
+  // variables
+
+  CephContext* cct;
+  std::map<std::string, Data1> data;
+
+public:
+
+  RGWArrowServer(const flt::FlightServerOptions& options, CephContext* _cct) :
+    cct(_cct)
+  {
+    arw::Status s = Init(options);
```

Comment:
where does all-caps Init(...) come from?

Reply:
That's a function inherited from the base class in Arrow Flight.